lib/src/config.dart

```diff
@@ -176,20 +200,23 @@ void _parseDeletionPatterns(yaml, GlobalConfig config) {
       .addAll((patterns as YamlList).map((pattern) => new RegExp(pattern)));
 }
 
-void _parseFilesToLoad(yaml, GlobalConfig config) {
+Future _parseFilesToLoad(yaml, GlobalConfig config) async {
   var filePaths = _parseStringList(yaml, 'files_to_load');
   if (filePaths == null) return;
-  config.filesToLoad.addAll(filePaths.map((filePath) {
+  config.filesToLoad.addAll(await Future.wait(filePaths.map((filePath) async {
     var parts = filePath.split(':');
     if (parts.length == 1) {
       return platformIndependentPath(parts[0]);
     } else if (parts.length == 2 && parts[0] == 'package') {
-      return path.join('packages', platformIndependentPath(parts[1]));
+      Uri uri = await config.packageResolver.resolveUri(filePath);
+      String p = platformIndependentPath(uri.toFilePath());
+      p = p.replaceAll("%", r'%25'); // Reset url encoding
```

Comment:
This is how a `.packages` file encodes path when you have a `PUB_HOSTED_URL` that contains a port (a colon char) : 

    js:file:///home/dtrump/.pub-cache/hosted/http%2558%2547%2547pub.domain.it%25585001/js-0.6.1/lib/

for example. It gets escaped two times. That's why I had to unescape the `%` char.